src/clock.rs

```diff
@@ -39,7 +38,16 @@ impl Instant {
             (
                 InstantKind::Virtual { nanoseconds },
                 InstantKind::Virtual { nanoseconds: earlier },
-            ) => Duration::from_nanos(nanoseconds.saturating_sub(earlier)),
+            ) => {
+                // It is possible for second to overflow because u64::MAX < (u128::MAX / 1e9).
+                let seconds = u64::try_from(
+                    nanoseconds.saturating_sub(earlier).saturating_div(1_000_000_000),
```

Comment:
I think `nanoseconds.saturating_sub(earlier)` should be hoisted out of here. Both `seconds` and `nanosecond` should be based on that subtraction.

Reply:
Oops yes, that's an error.

Reply:
Thanks, I suspect the previous commit went into an infinite loop somewhere because this error.